docs/samples/Microsoft.ML.Samples/Dynamic/SimpleDataViewImplementation.cs

```diff
@@ -0,0 +1,258 @@
+        /// the first pass, given the immutability principle of <see cref="IDatView"/>), and the
+        /// <see cref="GetRowCursorSet(IEnumerable{DataViewSchema.Column}, int, Random)"/> method returns a single item.
+        /// 
+        /// The <see cref="DataViewRowCursor"/> derived class has more documentation specific to its behavior.
+        /// 
+        /// Note that this implementation, as well as the nested <see cref="DataViewRowCursor"/> derived class, does almost
+        /// no validation of parameters or guard against misuse than we would like from, say, implementations of the same
+        /// classes within the ML.NET codebase.
+        /// </summary>
+        private sealed class InputObjectDataView : IDataView
+        {
+            private readonly IEnumerable<InputObject> _data;
+            public DataViewSchema Schema { get; }
+            public bool CanShuffle => false;
+
+            public InputObjectDataView(IEnumerable<InputObject> data)
+            {
+                _data = data;
+
+                var builder = new DataViewSchema.Builder();
+                builder.AddColumn("Label", BooleanDataViewType.Instance);
+                builder.AddColumn("Text", TextDataViewType.Instance);
+                Schema = builder.ToSchema();
+            }
+
+            public long? GetRowCount() => null;
+
+            public DataViewRowCursor GetRowCursor(IEnumerable<DataViewSchema.Column> columnsNeeded, Random rand = null)
+            {
+                return new Cursor(this, columnsNeeded.Any(c => c.Index == 0), columnsNeeded.Any(c => c.Index == 1));
+            }
+
+            public DataViewRowCursor[] GetRowCursorSet(IEnumerable<DataViewSchema.Column> columnsNeeded, int n, Random rand = null)
+                => new[] { GetRowCursor(columnsNeeded, rand) };
+
+            /// <summary>
+            /// Having this be a private sealed nested class follows the typical pattern: in most <see
+            /// cref="IDataView"/>
+            /// implementations, the cursor instance is almost always that. The only "common" exceptions to this
+            /// tendency are those implementations that are such thin wrappings of existing <see cref="IDataView"/>
+            /// without even bothering to change the schema.
+            ///
+            /// On the subject of schema, note that there is an expectation that the <see cref="Schema"/> object is
+            /// reference equal to the <see cref="IDataView.Schema"/> object that created this cursor, as we see here.
+            ///
+            /// Note that <see cref="Batch"/> returns <c>0</c>. As described in the documentation of that property, that
+            /// is meant to facilitate the reconciliation of the partitioning of the data in the case where multiple
+            /// cursors are returned from
+            /// <see cref="GetRowCursorSet(IEnumerable{DataViewSchema.Column}, int, Random)"/>, but since only one is
```

Comment:
nit: fix new line.

Reply:
Sigh... @artidoro do you happen to know of any tool that does line breaking well without doing *this* stuff?